Add recursion depth for `FeatureTreeNode::getChildren`

s9latimm opened the pull request:
Add a depth parameter and update documentation.

s9latimm committed:
Add recursion depth

M	include/vara/Feature/FeatureTreeNode.h
M	unittests/Feature/Feature.cpp
M	unittests/Feature/Relationship.cpp

vulder commented on include/vara/Feature/FeatureTreeNode.h:
```diff
@@ -76,10 +76,19 @@ class FeatureTreeNode : private llvm::DGNode<FeatureTreeNode, FeatureTreeNode> {
+  ///  children of this node are considered and disables expansions.
+  ///
+  /// \return Set of children nodes.
+  template <typename T> llvm::SmallSet<T *, 3> getChildren(int Depth = -1) {
```
Please add a default template parameter to FeatureTreeNode

vulder reviewed (changes requested):
Could you please add 1-2 tests for edge cases, like when the depth given is deeper than the tree itself.

vulder committed:
Merge branch 'vara-dev' into f-GetChildrenDepth

s9latimm committed:
Add default template and tests

M	include/vara/Feature/FeatureTreeNode.h
M	unittests/Feature/CMakeLists.txt
M	unittests/Feature/Feature.cpp
A	unittests/Feature/FeatureTreeNode.cpp
M	unittests/Feature/Relationship.cpp
